cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ProgramGenerationStage.java

```diff
@@ -60,18 +60,6 @@ public void process(final ApplicationDeployable input) throws Exception {
     List<ProgramDescriptor> programDescriptors = new ArrayList<>();
     final ApplicationSpecification appSpec = input.getSpecification();
 
-    // Make sure the namespace directory exists
```

Comment:
yeah, that's what terence's comment is about..  i was able to test by deleting `programs` dir and deploying an app and that works.  well terence asked if it creates anything, it will create `programs` dir on a new namespace, but isn't probably necessary as it worked without that.